mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -246,4 +242,127 @@ private static MlflowHostCredsProvider getHostCredsProviderFromTrackingUri(Strin
     }
     return provider;
   }
+
+  /**
+   * Uploads the given local file to the run's root artifact directory. For example,
+   *
+   *  logArtifact(runId, "/my/localModel")
+   *  listArtifacts(runId) // returns "localModel"
+   *
+   * @param runId Run ID of an existing MLflow run.
+   * @param localFile File to upload. Must exist, and must be a simple file (not a directory).
+   */
+  public void logArtifact(String runId, File localFile) {
+    getArtifactRepository(runId).logArtifact(localFile);
+  }
+
+  /**
+   * Uploads the given local file to an artifactPath within the run's root directory. For example,
+   *
+   *   logArtifact(runId, "/my/localModel", "model")
+   *   listArtifacts(runId, "model") // returns "model/localModel"
+   *
+   * (i.e., the localModel file is now available in model/localModel).
+   *
+   * @param runId Run ID of an existing MLflow run.
+   * @param localFile File to upload. Must exist, and must be a simple file (not a directory).
+   * @param artifactPath Artifact path relative to the run's root directory. Should NOT
+   *                     start with a /.
+   */
+  public void logArtifact(String runId, File localFile, String artifactPath) {
```

Comment:
We should consider whether we want to use File or String for local paths here. Which will be more natural in Java? File is more type-safe and you might get it from various Java APIs, but it's also a bit of a departure from the Python and R APIs that might make the API more verbose. Maybe it's fine though.

Reply:
As discussed offline, File is pretty common w/in Java APIs to represent, specifically, files or directories on the local file system. And having a non-String arguments makes it less likely to flip arguments (as you can't do call-by-name in Java unlike Python).